### PR TITLE
Renamed global paths getters

### PR DESCRIPTION
**Pull request by igorkorsukov**
And added page about resources to wiki: https://github.com/musescore/MuseScore/wiki/Resources

### CODE REVIEW

File: src/extensions/internal/extensionsconfiguration.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ void ExtensionsConfiguration::init()
 {
     settings()->setDefaultValue(CHECK_FOR_UPDATE, Val(true));
 
-    settings()->setDefaultValue(USER_EXTENSIONS_PATH, Val(globalConfiguration()->sharePath().toStdString() + EXTENSIONS_DIR.toStdString()));
+    settings()->setDefaultValue(USER_EXTENSIONS_PATH, Val((globalConfiguration()->appDataPath() + EXTENSIONS_DIR).toStdString()));
```
Comment (igorkorsukov) [inline src/extensions/internal/extensionsconfiguration.cpp:56]: @Eism It looks like there is a mistake. The key is called `user...` and the application data folder is used (not the user data folder)

**Comment by cbjeukendrup** [comment]
For MuseScore 3, on macOS it was like this: 
* `mscore.app/Contents/Resources/`
    * (app data)
    * NOTE: is Read-only, on macOS!!
* `~/Documents/MuseScore[Version]/`
    * Extensions/
    * Images/
    * Plugins/
    * Scores/
    * SoundFonts/
    * Styles/
    * Templates/
* `~/Library/Application Support/MuseScore/MuseScore[Version]/`
    * locale/ = downloaded translations
    * plugins/ = ??
    * QtWebEngine/ = ??
    * tours/ 
        * completedTours.list
    * workspaces/
    * session
    * shortcuts.xml
* `~/Library/Preferences/org.musescore.MuseScore[Version].plist`

I think it's okay to have the preferences file in that place (that's very common for macOS apps). 

About the other user data paths:
I think there are two things. 
A path like `~/Documents/MuseScore[Version]/` for "visible" user data, like templates, style files, SoundFonts; things the user installs or edits.
And a path like `~/Library/Application Support/MuseScore/MuseScore[Version]/` for "hidden" data, e.g. data that will/should not be displayed in its raw form to the user, like which tours are completed, session files, and auto-save files.

In MuseScore 3, shortcuts.xml is also in the "hidden" user data folder. I think this is good, because it's bad if an uneducated user edits this file and corrupts it, and because the values inside it can also be edited via the GUI. 

When the user wants to share this file, it can be "exported" in the Preferences dialog, so that the user can save a copy of the file somewhere they choose, and then also imported.

File: src/extensions/internal/extensionsconfiguration.cpp (hunk 1 above)
Comment (cbjeukendrup) [reply src/extensions/internal/extensionsconfiguration.cpp:56]: I think it should indeed be "userDataPath", and also a `+ "/" ` is missing.

**Comment by igorkorsukov** [comment]
> For MuseScore 3, on macOS it was like this:
> 
> * `mscore.app/Contents/Resources/`
>   
>   * (app data)
>   * NOTE: is Read-only, on macOS!!
> * `~/Documents/MuseScore[Version]/`
>   
>   * Extensions/
>   * Images/
>   * Plugins/
>   * Scores/
>   * SoundFonts/
>   * Styles/
>   * Templates/
> * `~/Library/Application Support/MuseScore/MuseScore[Version]/`
>   
>   * locale/ = downloaded translations
>   * plugins/ = ??
>   * QtWebEngine/ = ??
>   * tours/
>     
>     * completedTours.list
>   * workspaces/
>   * session
>   * shortcuts.xml
> * `~/Library/Preferences/org.musescore.MuseScore[Version].plist`
> 
> I think it's okay to have the preferences file in that place (that's very common for macOS apps).
> 
> About the other user data paths:
> I think there are two things.
> A path like `~/Documents/MuseScore[Version]/` for "visible" user data, like templates, style files, SoundFonts; things the user installs or edits.
> And a path like `~/Library/Application Support/MuseScore/MuseScore[Version]/` for "hidden" data, e.g. data that will/should not be displayed in its raw form to the user, like which tours are completed, session files, and auto-save files.
> 
> In MuseScore 3, shortcuts.xml is also in the "hidden" user data folder. I think this is good, because it's bad if an uneducated user edits this file and corrupts it, and because the values inside it can also be edited via the GUI.
> 
> When the user wants to share this file, it can be "exported" in the Preferences dialog, so that the user can save a copy of the file somewhere they choose, and then also imported.

Thanks for the info, I started trying to figure out which path to what.

File: src/instruments/internal/instrumentsconfiguration.cpp [hunk 2]
```diff
@@ -48,7 +48,7 @@ void InstrumentsConfiguration::init()
     });
 
     settings()->setDefaultValue(FIRST_SCORE_ORDER_LIST_KEY,
-                                Val(globalConfiguration()->sharePath().toStdString() + "instruments/orders.xml"));
+                                Val(globalConfiguration()->appDataPath().toStdString() + "instruments/orders.xml"));
```
Comment (igorkorsukov) [inline src/instruments/internal/instrumentsconfiguration.cpp:51]: Looks like there is missing `/`

**Review by vpereverzev — APPROVED** [review]

**Pull request merged**